lib/system/strmantle.nim

```diff
@@ -49,22 +104,22 @@ proc addInt*(result: var string; x: int64) =
   ##     b = 45
   ##   a.addInt(b) # a <- "12345"
   let base = result.len
-  setLen(result, base + sizeof(x)*4)
-  var i = 0
-  var y = x
-  while true:
-    var d = y div 10
-    result[base+i] = chr(abs(int(y - d*10)) + ord('0'))
-    inc(i)
-    y = d
-    if y == 0: break
+  var length: int
+  var num: uint64
+
   if x < 0:
-    result[base+i] = '-'
-    inc(i)
-  setLen(result, base+i)
-  # mirror the string:
-  for j in 0..i div 2 - 1:
-    swap(result[base+j], result[base+i-j-1])
+    if x == low(int64):
+      num = uint64(x)
+    else:
+      num = uint64(-x)
+    length = base + digits10(num) + 1
+    setLen(result, length)
+    result[base] = '-'
+  else:
+    num = uint64(x)
+    length = base + digits10(num)
+    setLen(result, length)
+  numToString(result, num, length)
```

Comment:
simpler:
```nim
  if x < 0:
    if x == low(int64):
      num = uint64(x)
    else:
      num = uint64(-x)
    length = base + digits10(num) + 1
    result[base] = '-'
  else:
    num = uint64(x)
    length = base + digits10(num)
  setLen(result, length)
```

Reply:
Yeah, I forgot to remove that after fixing an issue.

Reply:
`result[base] = '-'` needs setLen